src/main/java/io/github/perrymant/moneymaker/FileIO.java

```diff
@@ -0,0 +1,38 @@
+package io.github.perrymant.moneymaker;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+class FileIO {
+
+    String read(String fileName) {
+        try {
+            return new String(Files.readAllBytes(Paths.get(fileName)), Charset.defaultCharset());
+        } catch (RuntimeException | IOException e) {
+            e.printStackTrace();
```

Comment:
We don't catch RuntimeException here, we throw it instead of `e.printStackTrace();`. Since it's a RuntimeException (or better yet, your own class that extends from it) we don't have to declare it in the signature of `read` and the implementation details of `FileIO` need not be known by classes that use `FileIO`.

Applies throughout.